lib/api/api_base.dart

```diff
@@ -0,0 +1,50 @@
+export 'dart:convert';
+export 'dart:typed_data';
+export 'package:gatecheck_frontend/utils/string_uuid_conversion.dart';
+
+
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+enum DataStatus { Success, Unauthorized, Failure }
+
+String api = '/api/v1';
+class ApiData<T> {
+  DataStatus status;
+  T data;
+
+  ApiData(T Data, DataStatus Status)
+      : data = Data,
+        status = Status;
+}
+
+
+///this function takes a response and a toJson constructor for a certain type and returns an ApiData object for the specified type.
+ApiData<T> parseResponse<T>(http.Response res, T toJson(dynamic json)){
+  if(res.statusCode==200){
+    var json=jsonDecode(res.body);
+    return ApiData(toJson(json), DataStatus.Success);
+  }
+
+  if (res.statusCode == 401) {
+    return ApiData<T>(null, DataStatus.Unauthorized);
+  }
+  return ApiData<T>(null, DataStatus.Failure);
```

Comment:
You would want to wrap this in a try catch since an error might be thrown.
Most likely you would want to return something like:
```
try {
    throw Exception("FDJfihsndiof");
  } catch (err, stacktrace) {
    ApiData<T>(stacktrace, DataStatus.Exeception)
  }```
  or some united error object with both err and stacktrace

Reply:
how will an error be thrown there?